### PR TITLE
Migrate models/addresses/*.sql and  models/addresses_events/*.sql

### PR DESCRIPTION
**Pull request by Hosuke**
Most addresses spells are static and have no dependencies.

### CODE REVIEW

File: macros/test-helpers/check_seed_macro_legacy.sql [hunk 1]
```diff
@@ -54,8 +54,8 @@
     matching_count_test as (
         select
             'matched records count' as test_description,
-            count(model_{{seed_matching_columns[0]}}) as `result_model`,
-            1 as `expected_seed`,
+            count(model_{{seed_matching_columns[0]}}) as result_model,
+            1 as expected_seed,
```
Comment (Hosuke) [inline macros/test-helpers/check_seed_macro_legacy.sql:58]: I think remove the ` in check_seed_macro_legacy to fix errors in dbt-test.

File: seeds/addresses_events/first_funded_by_seed.csv [hunk 2]
```diff
@@ -2,7 +2,7 @@ blockchain,address,first_funded_by,block_number,tx_hash
 gnosis,0x000000000000c1a8cc407293a22bff068d4584d2,0xbe51d38547992293c89cc589105784ab60b004a9,25390104,0x80f6ddbb341615b5ed83a2d1a3e51deb202edfffa24dd6a787c245db68bb6ed5
 gnosis,0x000000a52a03835517e9d193b3c27626e1bc96b1,0x115909bdcbab21954beb4ab65fc2abee9866fa93,21134610,0x75e2c7811c9c87def2fc113d5b01bf51e100fc449bda4cbdcec66d3b1215bd07
 goerli,0x1a32f88ca270e1092dac040ce6cf629fbc60d1b8,0xe73f10e30ba43e55dcb9844e9a91fc1c42ad0fea,6398877,0x92b47d7ef3dad447885d066e83851e7b3cbd0a9eef44c22d0d8dff3f5611631b
-goerli,0x1a332b3cf20466189c221a04f92e47c21a17ba19,0x5cf0e1f993131e12c5fdd3dfc8ba52b30fcdcc14,4449324,0x99866f8bb5e22538a11a67afc11cdae62e3e9c00506e9f8642f5819d1d5ec9e1
+goerli,0x1a332b3cf20466189c221a04f92e47c21a17ba19,0x5cf0e1f993131e12c5fdd3dfc8ba52b30fcdcc14,4449324,0xd64697d0f8b3455a40385b6d71170ee5fd65f7f7e833c3d2782eccaec10d809e
```
Comment (Hosuke) [inline seeds/addresses_events/first_funded_by_seed.csv:5]: Based on etherscan, there are 2 tx_hash for this same block_number:
https://goerli.etherscan.io/address/0x1a332b3cf20466189c221a04f92e47c21a17ba19

Comment (Hosuke) [reply seeds/addresses_events/first_funded_by_seed.csv:5]: And `0xd64697d0f8b3455a40385b6d71170ee5fd65f7f7e833c3d2782eccaec10d809e` may be the earlier transaction.

File: macros/test-helpers/check_seed_macro_legacy.sql (hunk 1 above)
Comment (jeff-dude) [reply macros/test-helpers/check_seed_macro_legacy.sql:58]: interesting, this was failing on the spark CI step? how did it run previously then?

File: models/addresses_events_testnets/goerli/addresses_events_testnets_goerli_first_funded_by.sql [hunk 3]
```diff
@@ -10,19 +11,21 @@
...
 , MIN(et.block_time) AS block_time
 , MIN(et.block_number) AS block_number
 , MIN_BY(et.tx_hash, et.block_number) AS tx_hash
 FROM {{ source('goerli', 'traces') }} et
 {% if is_incremental() %}
-LEFT ANTI JOIN {{this}} ffb ON et.to = ffb.address
+LEFT JOIN {{this}} ffb ON et.to = ffb.address
+WHERE ffb.address IS NULL
+{% else %}
+WHERE 1 = 1
```
Comment (jeff-dude) [inline models/addresses_events_testnets/goerli/addresses_events_testnets_goerli_first_funded_by.sql:23]: same idea here on the where clause moving

**Review by jeff-dude — CHANGES_REQUESTED** [review]

File: models/addresses_events/arbitrum/addresses_events_arbitrum_first_funded_by.sql [hunk 4]
```diff
@@ -10,19 +11,20 @@
...
+, MIN_BY(et."from", et.block_number) AS first_funded_by
 , MIN(et.block_time) AS block_time
 , MIN(et.block_number) AS block_number
 , MIN_BY(et.tx_hash, et.block_number) AS tx_hash
 FROM {{ source('arbitrum', 'traces') }} et
 {% if is_incremental() %}
-LEFT ANTI JOIN {{this}} ffb ON et.to = ffb.address
+LEFT JOIN {{this}} ffb ON et.to = ffb.address WHERE ffb.address IS NULL
+{% else %}
+WHERE 1 = 1
```
[the remark this answers is not in the document]
Comment (Hosuke) [reply models/addresses_events/arbitrum/addresses_events_arbitrum_first_funded_by.sql:22]: The left anti join is not available in dune sql engine, so I use `left join + where is null` to replace them. The solution involves a where clause. Do you have any other ideal solution for left anti join? I will adopt it globally.

File: macros/test-helpers/check_seed_macro_legacy.sql (hunk 1 above)
Comment (Hosuke) [reply macros/test-helpers/check_seed_macro_legacy.sql:58]: The failing error looks like this:
https://github.com/duneanalytics/spellbook/actions/runs/5574212027/jobs/10182523270#step:11:65

Comment (jeff-dude) [reply macros/test-helpers/check_seed_macro_legacy.sql:58]: that failure is happening on the dunesql CI step, but that macro is marked as `legacy`. it should be running on spark instead. you likely need to update your schema files to call the non-legacy seed macro, so it runs the correct query

Comment (jeff-dude) [reply macros/test-helpers/check_seed_macro_legacy.sql:58]: please follow example in #3804 for how to use seeds in migration. we should ultimately be able to remove changes to this legacy file from this PR.

File: models/addresses_events/arbitrum/addresses_events_arbitrum_first_funded_by.sql (hunk 4 above)
[the remark this answers is not in the document]
Comment (jeff-dude) [reply models/addresses_events/arbitrum/addresses_events_arbitrum_first_funded_by.sql:22]: nevermind, this is fine, sorry for the confusion

File: macros/test-helpers/check_seed_macro_legacy.sql (hunk 1 above)
Comment (Hosuke) [reply macros/test-helpers/check_seed_macro_legacy.sql:58]: I will recover this file.

**Review by jeff-dude — APPROVED** [review]
thank you ser 🙏

**Pull request merged**